backend/src/LibExecution/Interpreter.fs

```diff
@@ -240,6 +241,46 @@ let rec eval' (state : ExecutionState) (st : Symtable) (e : Expr) : DvalTask =
             return err id $"Missing key `{key}` in {typeStr}"
         | _ -> return result
 
+    | ERecordUpdate(id, baseRecord, updates) ->
+      let! baseRecord = eval state st baseRecord
+      match baseRecord with
+      | DRecord(typeName, _) ->
+        let typeStr = FQTypeName.toString typeName
+        let types = ExecutionState.availableTypes state
+        match recordMaybe typeName with
+        | None ->
+          let typ = Types.find typeName types
+          match typ with
+          | None -> return err id $"There is no type named `{typeStr}`"
+          | Some(CustomType.Enum _) ->
+            return err id $"Expected a record but {typeStr} is an enum"
+          | _ -> return err id $"Expected a record but {typeStr} is something else"
+        | Some(expected1, expectedRest) ->
+          let expectedFields =
+            (expected1 :: expectedRest) |> List.map (fun f -> f.name, f.typ) |> Map
+          return!
+            Ply.List.foldSequentially
+              (fun r (k, expr) ->
+                uply {
+                  let! v = eval state st expr
+                  match r, k, v with
+                  | r, _, _ when Dval.isFake r -> return r
+                  | _, _, v when Dval.isFake v -> return v
+                  | _, "", _ -> return err id $"Empty key for value `{v}`"
+                  | DRecord(_, m), _, _ when not (Map.containsKey k m) ->
```

Comment:
I think this should check against `expectedFields`?

Reply:
Definitely, `expectedFields` and `m` consist of the same values. By using the `m` I wanted to highlight the fact that we are looking for a field among **existing** fields. Plus, it is easier to find with the eyes `m` than `expectedFields`, so the cognitive load becomes lower :) Btw as an option we could rename `expectedFields` to `existedFields` and that would resolve our issue. And sure I can replace `m` on `expectedFields`, choose ;)

Reply:
I don't understand why `expectedFields` is the same as `m`? Isn't `expectedFields` made up of the type's fields, while `m` comes from the `DRecord`'s existing fields?

I'm thinking maybe you mean that we expect `m` and `expectedFields` to have the same fields to them, which I think we expect but is not guaranteed.

Reply:
Yes, you are right. I meant that both maps have the same keys, not values. They have to have common set of keys because they based on one common source i.e.`baseRecord`. And as we are checking keys (which we are updating), IMO it doesn't matter what a map we are choosing for that. Ok, or am I wrong?

Reply:
I'm thinking that it shouldn't matter in theory. However, it is possible to have bugs in the creation of `m`, but much less likely to have bugs in `expectedFields`. If there is a bug with missing fields, the error message will be weird and hard to debug if we use `m` - better to go with the source of truth, which is `expectedFields`.

Reply:
Indeed, agree. I've replaced `m` on `expectedFields`